File: 2013-06-07/python/exercise6.py
def lar_to_obj(larModel):
	vertices, facets = larModel
	s = ""
	for vert in vertices :
		if len(vert) == 2 :
			s += "v "+str(vert[0])+" "+str(vert[1])+" 0\n"
		else :
			if len(vert) == 3 : 
				s += "v "+str(vert[0])+" "+str(vert[1])+" "+str(vert[2])+"\n"
			else :
				s += "v "+str(vert[0])+" "+str(vert[1])+" "+str(vert[2])+" "+str(vert[3])+"\n"
	s += "\n"
	for facet in facets :
		s += "f "
		for el in facet :
			s += str(el)+" "
		s += "\n"
	return s

# example

FV = [[5,6,7,8],
[0,5,8],
[0,4,5],
[1,2,4,5],
[2,3,5,6],
[0,8,7], [3,6,7], [1,2,3], [0,1,4]
]

V = [[0,6],
[0,0],
[3,0],
[6,0],
[0,3],
[3,3],
[6,3],
[6,6],
[3,6]]

l = lar_to_obj([V, FV])

print(l)

File: 2013-06-07/javascript/exercise6.js
var lar_to_obj = function(larModel){
	var vertices = larModel[0];
	var facets = larModel[1];
	var s = "";
	for (var i = 0; i < vertices.length; i++){
		if (vertices[i][2] != null || typeof(vertices[i][2]) != "undefined")
			s += "v "+vertices[i][0]+" "+vertices[i][1]+" "+vertices[i][2]+" "+vertices[i][3]+"\n";
		else if (vertices[i][3] != null || typeof(vertices[i][3]) != "undefined") 
			s += "v "+vertices[i][0]+" "+vertices[i][1]+" 0 "+vertices[i][3]+"\n";
		else s += "v "+vertices[i][0]+" "+vertices[i][1]+" 0 "+"\n";
	}
	s += "\n";
	for (var i = 0; i < facets.length; i++){
		s+="f "
		facets[i].forEach(function(el){
			s += el.toString()+" ";
		});
		s+="\n";
	}
	return(s);
}

